K2Bridge/Utils/DateMathParser.cs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+namespace K2Bridge.Visitors
+{
+    using System.Text.RegularExpressions;
+
+    /// <content>
+    /// Parser for the Date Math expressions.
+    /// </content>
+    public class DateMathParser
+    {
+        public static string ParseDateMath(string expr)
+        {
+            // See Wiki or ES docs for details on date math syntax
+            var pattern = @"^(?<anchor>now|.+?(?:\|\||$))(?<ranges>(?:(?:\+|\-)[^\/]*))?(?<rounding>\/(?:y|M|w|d|h|H|m|s))?$";
+
```

Comment:
Technically there could be multiple range operations.
( see https://www.elastic.co/guide/en/elasticsearch/client/net-api/current/date-math-expressions.html  "Including multiple operations")

Do we want to support it?

Reply:
Right, so we could have `2018-02-01||+1d-1m` for example. I don't know how frequent that use case is, to be frank I missed it in the docs! 

I have pushed an attempt at parsing this using group captures, let me know what you think.